nx/lib/nx/lin_alg.ex

```diff
@@ -38,10 +38,17 @@ defmodule Nx.LinAlg do
       >
   """
   defn adjoint(t, opts \\ []) do
-    transform(t, fn
-      %{type: {:c, _}} = t -> t |> Nx.transpose(opts) |> Nx.conjugate()
-      t -> Nx.transpose(t, opts)
-    end)
+    __adjoint__(t, opts)
+  end
+
+  deftransformp __adjoint__(t, opts) do
+    case t do
+      %{type: {:c, _}} ->
+        t |> Nx.transpose(opts) |> Nx.conjugate()
+
+      _ ->
+        Nx.transpose(t, opts)
+    end
   end
```

Comment:
I wonder what would be the consequences of exposing `deftransform adjoint` instead of `defn adjoint`

Reply:
Same with `norm`, below

Reply:
In this case, none, we can probably do it today. In general, the main differences are:

1. Inputs are not converted into expressions
2. and therefore they are not jitted

But given it is a small function, it is fine. We should probably add those notes to `deftransformer` docs.